nixos/modules/services/misc/moonraker.nix

```diff
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.moonraker;
+  format = pkgs.formats.ini {
+    listToValue = l: "\n" + builtins.concatStringsSep "\n" (map (elem: "  ${elem}") l);
+    mkKeyValue = generators.mkKeyValueDefault {} ":";
+  };
+in {
+  options = {
+    services.moonraker = {
+      enable = mkEnableOption "Moonraker, an API web server for Klipper";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.moonraker;
+        description = "The Moonraker package.";
+      };
+
+      klipperSocket = mkOption {
+        type = types.path;
+        example = "/run/klipper/api";
+        description = "Path to Klipper's API socket.";
+      };
+
+      stateDir = mkOption {
+        type = types.path;
+        default = "/var/lib/moonraker";
+        description = "The directory containing the Moonraker databases.";
+      };
+
+      configDir = mkOption {
+        type = types.path;
+        default = cfg.stateDir + "/config";
+        description = ''
+          The directory containing client-writable configuration files.
+
+          Clients will be able to edit files in this directory via the API. This directory must be writable.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "moonraker";
+        description = "User account under which Moonraker runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "moonraker";
+        description = "Group account under which Moonraker runs.";
+      };
+
+      listenHost = mkOption {
```

Comment:
This name strikes me as odd, this is usually just called `address` for other services.